alpa/device_mesh.py

```diff
@@ -1404,14 +1378,20 @@ def save(self, path: str):
         for host_id, uuids in buf_refs_per_host.items():
             if len(uuids) > 0:
                 obj_refs.append(
-                    self.device_mesh.workers[host_id].save_buffers_to_ts.remote(
+                    self.device_mesh.workers[host_id].save_buffers.remote(
                         path, uuids, indices_per_host[host_id], self.shape))
-        return ray.get(obj_refs)
+        if synchronized:
+            return None
+        else:
+            return obj_refs
```

Comment:
You do not need to return futures here. If users want to sync or wait for this buffer. They can call `executable.sync()`, `device_mesh.sync_workers()` or `array.block_until_ready()`